predicators/pybullet_helpers/robots/single_arm.py

```diff
@@ -429,7 +433,11 @@ def inverse_kinematics(self, end_effector_pose: Pose3D,
             joint_positions = self._ikfast_inverse_kinematics(
                 end_effector_pose)
             if validate:
-                self._validate_joints_state(joint_positions, end_effector_pose)
+                try:
+                    self._validate_joints_state(joint_positions,
+                                                end_effector_pose)
+                except ValueError as e:
+                    raise InverseKinematicsError(e)
             return joint_positions
 
         return pybullet_inverse_kinematics(
```

Comment:
Pybullet IK may also fail (in practice it sometimes just returns nonsense), in which case it just raises an Exception. So should we change that to `InverseKinematicsError`?

https://github.com/Learning-and-Intelligent-Systems/predicators/blob/b69599a0bb9a0bd6d4051a4e163ab3867420fdbb/predicators/pybullet_helpers/inverse_kinematics.py#L83